src/Control/Carrier/Choose/Church.hs

```diff
@@ -91,10 +93,10 @@ instance MonadTrans ChooseC where
   {-# INLINE lift #-}
 
 instance Algebra sig m => Algebra (Choose :+: sig) (ChooseC m) where
-  alg (hdl :: forall x . ctx (n x) -> ChooseC m (ctx x)) sig (ctx :: ctx ()) = case sig of
-    L Choose -> ChooseC $ \ fork leaf -> fork (leaf (True <$ ctx)) (leaf (False <$ ctx))
-    R other  -> ChooseC $ \ fork leaf -> thread dst hdl other (pure ctx) >>= runIdentity . runChoose (coerce fork) (coerce leaf)
+  alg hdl sig ctx = ChooseC $ \ fork leaf -> case sig of
+    L Choose -> leaf (True <$ ctx) `fork` leaf (False <$ ctx)
+    R other  -> thread (dst ~<~ hdl) other (pure ctx) >>= runIdentity . runChoose (coerce fork) (coerce leaf)
```

Comment:
Our various carriers wrapping functions have been refactored to float the functions out for readability.